### PR TITLE
[SD-2840] - (Refactoring) Image crop functionality

### PR DESCRIPTION
**Pull request by sjunaid**

### CODE REVIEW

**Comment by akintolga** [comment]
:+1:

File: client/app/scripts/superdesk-authoring/authoring.js [hunk 1]
```diff
@@ -687,71 +688,60 @@
...
+        $scope.data.cropData = {};
+        $scope.data.isDirty = false;
+
+        $scope.$watch('preview', function(newValue, oldValue) {
+            if (newValue === oldValue) {
+                $scope.origPreview = $scope.preview;
+                return;
             }
-        };
+            if (Object.keys(newValue).length === Object.keys(oldValue).length) {
```
Comment (plamut) [inline client/app/scripts/superdesk-authoring/authoring.js]: I suggest adding an explanation why this check is in place, it is probably not obvious for people not knowing all the details.

File: client/app/scripts/superdesk-authoring/authoring.js [hunk 2]
```diff
@@ -687,71 +688,60 @@
...
+                .then(angular.bind(this, function record() { // Ok = continue w/o saving
+                    $scope.data.isDirty = false;
+                    $scope.preview = $scope.origPreview;
+                    $scope.resolve($scope.data);
+                }));
             } else {
-                //Hold crop changes if item is already published, so it will be save on correction.
-                recordCrop(coordinates, cropName);
-                notify.success(gettext('Crop changes recorded'));
+                $scope.reject();
```
Comment (plamut) [inline client/app/scripts/superdesk-authoring/authoring.js]: Where do these `$scope.resolve()` and `$scope.reject()` methods come from?

File: client/app/scripts/superdesk-authoring/authoring.js [hunk 3]
```diff
@@ -687,71 +688,60 @@
...
+                recordCrops(cropsize.name);
+            });
+            notify.success(gettext('Crop changes have been recorded'));
+            $scope.resolve($scope.data);
+        };
+
+        $scope.close = function() {
+            if ($scope.data.isDirty) {
+                modal.confirm(gettext('You have unsaved changes, do you want to continue?'))
+                .then(angular.bind(this, function record() { // Ok = continue w/o saving
```
Comment (plamut) [inline client/app/scripts/superdesk-authoring/authoring.js]: Seems like binding the function to `this` is redundant here? The `record()` function doesn't use `this` anywhere in its body...

**Comment by plamut** [comment]
I would add a few function/method docstrings where missing, but apart from a few minor comments it looks good.

:+1:

**Comment by mdhaman** [comment]
:+1:  Looks good apart from the comments from @plamut

File: client/app/scripts/superdesk-authoring/authoring.js (hunk 2 above)
Comment (sjunaid) [inline client/app/scripts/superdesk-authoring/authoring.js]: $scope.locals, $scope.resolve() and $scope.reject() come from activity.

**Pull request merged**